engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2ClientApi.scala

```diff
@@ -28,12 +28,20 @@ class OAuth2ClientApi[ProfileResponse: Decoder, AccessTokenResponse: Decoder]
       "redirect_uri" -> configuration.redirectUrl
     ) ++ configuration.accessTokenParams
 
-    basicRequest
-      .body(payload.asJson)
-      .contentType(contentType.toString)
+    var request =
+      basicRequest
+      .contentType(configuration.accessTokenRequestContentType)
       .response(asJson[AccessTokenResponse])
       .post(Uri(configuration.accessTokenUri))
       .headers(configuration.headers)
+
+    val contentType = MediaType.parse(configuration.accessTokenRequestContentType)
+    request = contentType match {
+      case Right(MediaType.ApplicationJson) => request.body(payload.asJson)
+      case _ => request.body(payload)
```

Comment:
Will it work for all cases? Or just for x-www-form-urlencoded? Maybe we should explicitly enumerate types we can handle and throw exception otherwise?